ygot/render_test.go

```diff
@@ -2903,3 +2903,95 @@ func TestFindUpdatedLeaves(t *testing.T) {
 		})
 	}
 }
+
+func TestEmitRFC7951(t *testing.T) {
+	tests := []struct {
+		desc             string
+		in               interface{}
+		inArgs           *RFC7951JSONConfig
+		want             interface{}
+		wantErrSubstring string
+	}{{
+		desc: "simple string ptr field",
+		in:   String("test-string"),
+		want: `"test-string"`,
+	}, {
+		desc:             "scalar string - unsupported type",
+		in:               "invalid-scalar-string",
+		wantErrSubstring: "unexpected field type",
+	}, {
+		desc: "simple GoStruct",
+		in: &renderExample{
+			Str: String("test-string"),
+		},
+		want: `{"str":"test-string"}`,
+	}, {
+		desc: "map of GoStructs",
+		in: map[string]*renderExample{
+			"one": {Str: String("one")},
+			"two": {Str: String("two")},
+		},
+		want: `[{"str":"one"},{"str":"two"}]`,
+	}, {
+		desc: "map of invalid type",
+		in: map[string]string{
+			"one": "two",
+		},
+		wantErrSubstring: "invalid GoStruct",
+	}, {
+		desc: "map of invalid GoStruct",
+		in: map[string]*invalidGoStructField{
+			"one": {Value: "one"},
+		},
+		wantErrSubstring: "got unexpected field type",
+	}, {
+		desc: "slice of structs",
+		in: []*renderExample{
+			{Str: String("one")},
+		},
+		want: `[{"str":"one"}]`,
+	}, {
+		desc: "slice of scalars",
+		in:   []string{"one", "two"},
+		want: `["one","two"]`,
+	}, {
+		desc: "slice of annotations",
+		in: []*testAnnotation{
+			{
+				AnnotationFieldOne: "test",
+			},
+		},
+		want: `[{"field":"test"}]`,
+	}, {
+		desc: "empty annotation slice",
+		in:   []*testAnnotation{},
+		want: `null`,
+	}, {
+		desc: "empty map",
+		in:   map[string]*renderExample{},
+		want: `null`,
+	}, {
+		desc: "nil string pointer",
+		in: func() *string {
+			var s *string
+			return s
+		}(),
+		want: `null`,
+	}}
```

Comment:
Add test for empty type

```suggestion
        }, {
                desc: "empty type",
                in: &renderExample{
                        Empty: true,
                },
                want: `{"empty":[null]}`,
	}}
```

Reply:
Great, thanks! Added.